common/lib/capa/capa/capa_problem.py

```diff
@@ -808,7 +825,13 @@ def _extract_html(self, problemtree):  # private
         # otherwise, render children recursively, and copy over attributes
         tree = etree.Element(problemtree.tag)
         for item in problemtree:
-            item_xhtml = self._extract_html(item)
+            item_xhtml = self._extract_html(item, question_id)
+
+            if item.tag == 'question':
+                item_xhtml.set('class', 'question')
+                item_xhtml.set('id', str(question_id))
```

Comment:
@cptvitamin I have already changed the id value and now they will be `question-0`, `question-1` and so on.  Sorry i am unable to understand what do you mean by template inheritance related to IDrefs?